service/internal/obsconsumer/logs.go

```diff
@@ -0,0 +1,55 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package obsconsumer // import "go.opentelemetry.io/collector/service/internal/obsconsumer"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+var _ consumer.Logs = Logs{}
+
+func NewLogs(consumer consumer.Logs, itemCounter metric.Int64Counter, opts ...Option) Logs {
```

Comment:
given we'll need to record multiple metrics here in the future, would it make sense to set these via the options? i.e. there's a way to set:
* consumedCounter
* producedCounter
* consumedSize
* producedSize

and by default they're set to no-op instruments or wrapped in a nil check?

Reply:
i see, yeah i wasn't sure how this was meant to be used, but now i see. thanks for the explanation above.